lib/semantics/semantics.cc

```diff
@@ -203,34 +203,41 @@ void SemanticsContext::PopConstruct() {
   constructStack_.pop_back();
 }
 
-void SemanticsContext::SayDoVarRedefine(
-    const parser::CharBlock &location, const Symbol &variable) {
-  const parser::CharBlock doLoc{GetDoVariableLocation(variable)};
-  CHECK(doLoc != parser::CharBlock{});
-  Say(location, "Cannot redefine DO variable '%s'"_err_en_US, variable.name())
-      .Attach(doLoc, "Enclosing DO construct"_en_US);
-}
-
-void SemanticsContext::CheckDoVarRedefine(
-    const Symbol &variable, const parser::CharBlock &location) {
+void SemanticsContext::CheckDoVarRedefine(const parser::CharBlock &location,
+    const Symbol &variable, const parser::MessageFixedText message) {
```

Comment:
Slightly better to use `parser::MessageFixedText &&message` and then `std::move` it where it is used.

Reply:
Thanks, Tim.  I'll change this.